astropy/io/fits/fitsrec.py

```diff
@@ -803,18 +803,18 @@ def _convert_p(self, column, field, recformat):
             )
 
         for idx in range(len(self)):
-            offset = field[idx, 1] + self._heapoffset
+            offset = int(field[idx, 1]) + self._heapoffset
             count = field[idx, 0]
```

Comment:
My sense would be to do `count =  int(field[idx, 0])` here, so that it is more obvious why it is done (obviously, can then remove the two below.

Reply:
even better. Done !